ci/build.sh

```diff
@@ -161,5 +168,40 @@ case "${OS}" in
 esac
 
 for TARGET in $TARGETS; do
-    test_target "$TARGET"
+    test_target cargo "$TARGET"
 done
+
+# FIXME: https://github.com/rust-lang/rust/issues/58564
+# sparc-unknown-linux-gnu
+RUST_LINUX_NO_CORE_TARGETS="\
+x86_64-unknown-dragonfly \
+aarch64-pc-windows-msvc \
+aarch64-unknown-cloudabi \
+armv7-unknown-cloudabi-eabihf \
+i586-pc-windows-msvc \
+i686-pc-windows-gnu \
+i686-pc-windows-msvc \
+i686-unknown-cloudabi \
+i686-unknown-haiku \
+i686-unknown-netbsd \
+nvptx64-nvidia-cuda \
+powerpc-unknown-linux-gnuspe \
+riscv32imac-unknown-none-elf \
+riscv32imc-unknown-none-elf \
+sparc64-unknown-netbsd \
+thumbv8m.main-none-eabi \
+x86_64-pc-windows-gnu \
+x86_64-pc-windows-msvc
+x86_64-unknown-bitrig \
+x86_64-unknown-haiku \
+x86_64-unknown-openbsd
+"
+
+if [ "${RUST}" = "nightly" ]; then
+    rustup component add rust-src || true
+    cargo install xargo || true
```

Comment:
Why the `|| true`?  Won't that just lead to confusing error messages, for example when the internet isn't available?

Reply:
Without any of these the first `xargo` invocation right after will fail hard, so the error won't be far away (not as far as, e.g., when running tests under qemu inside a Docker container). 

OTOH if you have `xargo` installed, `cargo install xargo` errors, the same applies to `rust-src`, so this is the easiest way of supporting that, which allows you to run this locally without much issues.

Reply:
So i've removed the `|| true`, making this more reliable.